api/v1/submissions/PKPSubmissionFileController.php

```diff
@@ -320,12 +320,12 @@ public function add(Request $illuminateRequest): JsonResponse
         $params['submissionId'] = $submission->getId();
         $params['uploaderUserId'] = (int) $request->getUser()->getId();
 
-        $primaryLocale = $request->getContext()->getPrimaryLocale();
-        $allowedLocales = $request->getContext()->getData('supportedSubmissionLocales');
+        $submissionLocale = $submission->getData('locale');
+        $allowedLocales = $request->getContext()->getSupportedSubmissionMetadataLocales();
```

Comment:
@jyhein, do you know where is this function used? -- I wanted to test, but I do not know...

Reply:
I left this comment for me, to investigate further... :-)